aten/src/ATen/core/CachingHostAllocator.h

```diff
@@ -651,24 +673,54 @@ struct CachingHostAllocatorInterface : public at::Allocator {
     impl_->copy_data(dest, src, count);
   }
 
-  HostStats getStats() {
-    return impl_->getStats();
+  HostStats get_stats() override {
+    return impl_->get_stats();
   }
 
-  void resetAccumulatedStats() {
-    impl_->resetAccumulatedStats();
+  void reset_accumulated_stats() override {
+    impl_->reset_accumulated_stats();
   }
 
-  void resetPeakStats() {
-    impl_->resetPeakStats();
+  void reset_peak_stats() override {
+    impl_->reset_peak_stats();
   }
 
   std::unique_ptr<T> impl_;
 };
 
-#define DECLARE_HOST_ALLOCATOR(name, impl, deleter) \
-  struct name final                                 \
-      : public at::CachingHostAllocatorInterface<impl, deleter> {};
+#define DECLARE_HOST_ALLOCATOR(name, impl, deleter, instance)       \
+  void deleter(void* ptr);                                          \
+  struct name final                                                 \
+      : public at::CachingHostAllocatorInterface<impl, deleter> {}; \
+  static name instance;                                                    \
+  void deleter(void* ptr) {                                         \
+    instance.free(ptr);                                             \
+  }
+
+/**
+ * Set the host allocator for DeviceType `device_type`. This allocator manages
+ * pinned memory on the host that can be accessed efficiently by the specified
+ * device type. Note that this function is not thread-safe.
+ */
+TORCH_API void setHostAllocator(
+    at::DeviceType device_type,
+    at::HostAllocator* allocator,
+    uint8_t priority = 0);
+
+TORCH_API at::HostAllocator* getHostAllocator(at::DeviceType device_type);
```

Comment:
Do we really need a per-device lookup here?
Can we rely on the fact that there is a single accelerator right now?

Reply:
Sounds reasonable.   @guangyey  the default accelerator is able to tell which device type should be now?

Reply:
I think a per-device lookup design is safer.
From an implementation perspective, I recall that devices like `MTIA` or `PrivateUser1` can co-exist with a CUDA-enabled binary, since getAccelerator() is resolved at runtime rather than at build time for them.
In practice, users can combine `getAccelerator()` with `getHostAllocator()` to enable better interoperability across different backends.

Reply:
I guess it is, ok!